lib/galaxy/datatypes/registry.py

```diff
@@ -314,23 +319,49 @@ def append_to_sniff_order():
                     self.sniff_order.append( datatype )
         append_to_sniff_order()
 
-    def load_build_sites( self, root ):
+    def _load_build_sites( self, root ):
+
+        def load_build_site( build_site_config ):
+            # Take in either an XML element or simple dictionary from YAML and add build site for this.
+            if not (build_site_config.get( 'type' ) and build_site_config.get( 'file' )):
+                self.log.exception( "Site is missing required 'type' and 'file' attributes: %s" )
+                return
+
+            site_type = build_site_config.get( 'type' )
+            file = build_site_config.get( 'file' )
```

Comment:
It may be a good time to use a variable name which is not a Python2 reserved word.